### ISSUE TITLE
Modal window closes unexpectedly when 'mousedown' IN the modal and 'mouseup' on the background

### ISSUE THREAD

**Issue by DimaGashko** [ghosh/Micromodal#287]
Here's a video: https://drive.google.com/file/d/150mGwQNYmXyfykiVVdkw1VhCKvBhfQ96/view

This is small but really annoying issue!

p.s. the easiest solution is replace 'click' event by 'mousedown' (probably)

**Comment by piotrpog** [comment]
+1 for this issue. Honestly, is there modal lib that gets that right?

Its really annoying if there is form in the modal, users often click from input and release mouse button outside modal which closes it!

**Comment by PeterDekkers** [comment]
Duplicate of https://github.com/ghosh/Micromodal/issues/216

Worth mentioning that as a workaround you can disable clicking outside the modal to close by removing the `data-micromodal-close` attribute from the overlay.

I.e. change this:
```html
<div class="modal__overlay" tabindex="-1" data-micromodal-close>
```
To this:
```html
<div class="modal__overlay" tabindex="-1">
```

**Comment by PeterDekkers** [comment]
Also my 2 cents regarding the suggested fix of using the `mousedown` event. I guess this should _only_ be done for the overlay. In all other situations—e.g. close button—the `click` event would be correct.

So possibly this means the overlay should have a new `data-micromodal-mousedown-close` or some-such that listens to `mousedown` events.

Happy to make a PR if that's useful.

**Comment by DimaGashko** [comment]
@PeterDekkers yeah, removing data-micromodal-close is a the solution sometimes, but I need this functionality too.

mousedown - probably, but it should be tested on mobile and other touch devices also

**Comment by PeterDekkers** [comment]
Yes indeed @DimaGashko, good point about touch devices.

**Comment by rmkane** [comment]
Any update on the status of this issue? Look like the close trigger is still shared among the close button and modal overlay.

**Comment by kalpeshsingh** [comment]
Hello folks!
I don't have bandwidth to work on this.

Please send PR if possible and @ghosh or I can merge it.

**Comment by dkniffin** [comment]
If anyone has time to make a PR for this, I'd be happy to review it. Otherwise, I'll get to it when I can.

**Comment by robinstewart** [comment]
Expanding on https://github.com/micromodal/Micromodal/issues/505#issuecomment-1610172087 ...

It seems that the simplest workaround is to change the div structure to make the **container a sibling of the overlay** (instead of a child). Like this:

```
<div class="modal__root micromodal-slide" id="modal-1" aria-hidden="true">
    <div class="modal__overlay" tabindex="-1" data-micromodal-close></div>
    <div class="modal__container" role="dialog" aria-modal="true" aria-labelledby="${modalId}-title">
        <header class="modal__header">
            <h2 class="modal__title"></h2>
            <button class="modal__close" aria-label="Close modal" data-micromodal-close></button>
        </header>
        <main class="modal__content"></main>
        <footer class="modal__footer"></footer>
    </div>
</div>
```

If you are using the demo CSS, you will also need to change a few CSS properties at the root, overlay, and container level so the layout is correct.

See also #490
